crates/templates/src/git.rs

```diff
@@ -46,3 +46,24 @@ impl UnderstandGitResult for Result<std::process::Output, std::io::Error> {
         }
     }
 }
+
+pub(crate) async fn is_in_git_repo(dir: &Path) -> anyhow::Result<bool> {
+    let mut cmd = tokio::process::Command::new("git");
+    cmd.arg("-C")
+        .arg(dir)
+        .arg("rev-parse")
+        .arg("--git-dir")
+        .stdout(Stdio::null())
+        .stderr(Stdio::null());
+
+    let status = cmd.status().await?;
```

Comment:
nit: use of `context` would likely help here as the error message from the status call is likely to not be very helpful out of context.